Missing binding function with vague error

```
Caused by: java.lang.RuntimeException: java.util.NoSuchElementException: No value present
    at io.nosqlbench.virtdata.core.bindings.VirtDataFunctionResolver.resolveFunctions(VirtDataFunctionResolver.java:134)
    at io.nosqlbench.virtdata.core.bindings.VirtDataLibraries.resolveFunctions(VirtDataLibraries.java:52)
    at io.nosqlbench.virtdata.core.bindings.VirtDataComposer.resolve(VirtDataComposer.java:143)
    at io.nosqlbench.virtdata.core.bindings.VirtDataComposer.resolveDiagnosticFunctionFlow(VirtDataComposer.java:173)
    at io.nosqlbench.virtdata.core.bindings.VirtDataComposer.resolveFunctionFlow(VirtDataComposer.java:222)
    at io.nosqlbench.virtdata.core.bindings.VirtData.getOptionalMapper(VirtData.java:107)
    at io.nosqlbench.virtdata.core.templates.StringCompositor.lambda$new$1(StringCompositor.java:52)
    at java.base/java.util.HashMap.forEach(HashMap.java:1421)
    at io.nosqlbench.virtdata.core.templates.StringCompositor.<init>(StringCompositor.java:51)
    at io.nosqlbench.virtdata.core.templates.StringCompositor.<init>(StringCompositor.java:73)
    at io.nosqlbench.virtdata.core.templates.StringBindings.<init>(StringBindings.java:49)
    at io.nosqlbench.virtdata.core.templates.StringBindings.<init>(StringBindings.java:45)
    at io.nosqlbench.engine.api.templating.ParsedTemplateMap.lambda$applyTemplateFields$1(ParsedTemplateMap.java:142)
    at java.base/java.util.LinkedHashMap.forEach(LinkedHashMap.java:721)
    at io.nosqlbench.engine.api.templating.ParsedTemplateMap.applyTemplateFields(ParsedTemplateMap.java:123)
    at io.nosqlbench.engine.api.templating.ParsedTemplateMap.<init>(ParsedTemplateMap.java:113)
    at io.nosqlbench.adapters.api.templating.ParsedOp.<init>(ParsedOp.java:352)
    at io.nosqlbench.engine.api.activityimpl.uniform.StandardActivity.<init>(StandardActivity.java:92)
    at io.nosqlbench.engine.api.activityimpl.uniform.StandardActivityType.getActivity(StandardActivityType.java:62)
    at io.nosqlbench.engine.api.activityimpl.uniform.StandardActivityType.getActivity(StandardActivityType.java:38)
    at io.nosqlbench.engine.api.activityapi.core.ActivityType.getAssembledActivity(ActivityType.java:64)
```

--- a/adapter-cqld4/src/main/java/io/nosqlbench/engine/extensions/vectormath/VectorMath.java
+++ b/adapter-cqld4/src/main/java/io/nosqlbench/engine/extensions/vectormath/VectorMath.java
@@ -23,26 +23,47 @@ import java.util.List;
 
 public class VectorMath {
 
-    public static long[] rowsToLongArray(String fieldName, List<Row> rows) {
+    public static long[] rowFieldsToLongArray(String fieldName, List<Row> rows) {
         return rows.stream().mapToLong(r -> r.getLong(fieldName)).toArray();
+    }
+
+    public static String[] rowFieldsToStringArray(String fieldName, List<Row> rows) {
+        return rows.stream().map(r -> r.getString(fieldName)).toArray(String[]::new);
+    }
+
+    public static long[] stringArrayAsALongArray(String[] strings) {
+        long[] longs = new long[strings.length];
+        for (int i = 0; i < longs.length; i++) {
+            longs[i]=Long.parseLong(strings[i]);
+        }
+        return longs;
+    }
+
+    public static int[] stringArrayAsIntArray(String[] strings) {
+        int[] ints = new int[strings.length];
+        for (int i = 0; i < ints.length; i++) {
+            ints[i]=Integer.parseInt(strings[i]);
+        }
+        return ints;
     }
 
     public static int[] rowListToIntArray(String fieldName, List<Row> rows) {
         return rows.stream().mapToInt(r -> r.getInt(fieldName)).toArray();
     }
 
-    public double computeRecall(long[] referenceIndexes, long[] sampleIndexes) {
+    public static double computeRecall(long[] referenceIndexes, long[] sampleIndexes) {
         Arrays.sort(referenceIndexes);
         Arrays.sort(sampleIndexes);
         long[] intersection = Intersections.find(referenceIndexes,sampleIndexes);
         return (double)intersection.length/(double)referenceIndexes.length;
     }
 
-    public double computeRecall(int[] referenceIndexes, int[] sampleIndexes) {
+    public static double computeRecall(int[] referenceIndexes, int[] sampleIndexes) {
         Arrays.sort(referenceIndexes);
         Arrays.sort(sampleIndexes);
         int[] intersection = Intersections.find(referenceIndexes,sampleIndexes);
         return (double)intersection.length/(double)referenceIndexes.length;
     }
+
 
 }

--- a/adapter-cqld4/src/test/java/io/nosqlbench/engine/extensions/vectormath/VectorMathTest.java
+++ b/adapter-cqld4/src/test/java/io/nosqlbench/engine/extensions/vectormath/VectorMathTest.java
@@ -17,15 +17,13 @@
 package io.nosqlbench.engine.extensions.vectormath;
 
 import org.junit.jupiter.api.Test;
-
-import static org.junit.jupiter.api.Assertions.*;
 class VectorMathTest {
 
     private VectorMath vm = new VectorMath();
 
     @Test
     void computeRecallForRowListVsLongIndexList() {
-        new VectorMath();
+        VectorMath.computeRecall(new long[]{}, new long[]{});
     }
 
     @Test

--- a/virtdata-lib-hdf5/src/main/java/io/nosqlbench/virtdata/library/hdf5/from_long/AbstractHdfFileToVectorType.java
+++ b/virtdata-lib-hdf5/src/main/java/io/nosqlbench/virtdata/library/hdf5/from_long/AbstractHdfFileToVectorType.java
@@ -20,15 +20,13 @@ import io.jhdf.HdfFile;
 import io.jhdf.api.Dataset;
 import io.nosqlbench.api.content.NBIO;
 
-import java.nio.file.Paths;
-
 public abstract class AbstractHdfFileToVectorType {
     protected final HdfFile hdfFile;
     protected final Dataset dataset;
     protected final int[] dims;
 
     public AbstractHdfFileToVectorType(String filename, String datasetName) {
-        hdfFile = new HdfFile(NBIO.all().search(filename).first().get().asPath());
+        hdfFile = new HdfFile(NBIO.all().search(filename).one().asPath());
         //TODO: implement a function to get the dataset by name only without needing the full path
         dataset = hdfFile.getDatasetByPath(datasetName);
         dims = dataset.getDimensions();

